fish-rust/build.rs

```diff
@@ -103,6 +126,25 @@ fn main() {
     rsconf::rebuild_if_paths_changed(&source_files);
 }
 
+fn find_version() -> Result<String, Box<dyn std::error::Error>> {
```

Comment:
This appears to me to be what https://crates.io/crates/git-version says it does.

It just runs `git` itself, which I would find preferable over depending on the entirety of the git reimplementation that is libgit2.

Reply:
That's also basically [what we do now](https://github.com/fish-shell/fish-shell/blob/a29aa44183582d268ee2952e4eea050d0c96e1d1/build_tools/git_version_gen.sh)

Reply:
We could probably also just use the script directly (with `--stdout`). I don't know how the envisioned full replacement of CMake with Cargo will look like, and wanted this change to be somewhat independent of that

Reply:
I ended up just using the script directly

Reply:
Contra-message: To avoid change-detection issues, I re-added `git_version`, see https://github.com/fish-shell/fish-shell/pull/9962/commits/0ea11f5e7d5ae531cbc04e5f2846ba8145c49827